File: CheckFileCorruption/file_corruption_detector.py
import os
import subprocess

# Directory to scan
DIR = "/path/to/your/folder"

# Log file
LOGFILE = "corruption_log.txt"

# Supported movie formats
FORMATS = ["mp4", "mkv", "avi", "mov", "flv"]

# Clear the log file
with open(LOGFILE, "w") as log_file:
    log_file.write("")

# Function to check for corruption using Docker
def check_corruption(file):
    result = subprocess.run(
        [
            "docker", "run", "--rm", "-v",
            f"{os.path.abspath(DIR)}:/videos",
            "linuxserver/ffmpeg:latest",
            "-v", "error", "-i", f"/videos/{os.path.basename(file)}", "-f", "null", "-"
        ],
        stderr=subprocess.PIPE,
        stdout=subprocess.PIPE
    )
    with open(LOGFILE, "a") as log_file:
        if result.returncode != 0:
            log_file.write(f"Corruption detected in: {file}\n")
        else:
            log_file.write(f"No corruption detected in: {file}\n")

# Iterate through the directory
for format in FORMATS:
    for file in os.listdir(DIR):
        if file.endswith(f".{format}"):
            check_corruption(os.path.join(DIR, file))

print(f"Corruption check completed. Results are logged in {LOGFILE}.")

File: CheckFileCorruption/file_corruption_detector.rs
package main

import (
    "fmt"
    "io/ioutil"
    "log"
    "os"
    "os/exec"
    "path/filepath"
    "strings"
)

const (
    dir     = "/path/to/folder"
    logFile = "corruption_log.txt"
)

var formats = []string{"mp4", "mkv", "avi", "mov", "flv"}

func main() {
    f, err := os.Create(logFile)
    if err != nil {
        log.Fatal(err)
    }
    defer f.Close()

    files, err := ioutil.ReadDir(dir)
    if err != nil {
        log.Fatal(err)
    }

    for _, file := range files {
        if !file.IsDir() && isSupportedFormat(file.Name()) {
            fmt.Fprintf(f, "Filename: %s\n", file.Name())
            checkCorruption(file.Name(), f)
        }
    }

    fmt.Fprintf(f, "Corruption check completed. Results are logged in %s.\n", logFile)
}

func isSupportedFormat(filename string) bool {
    ext := strings.ToLower(filepath.Ext(filename))
    for _, format := range formats {
        if ext == "."+format {
            return true
        }
    }
    return false
}

func checkCorruption(file string, f *os.File) {
    cmd := exec.Command("docker", "run", "--rm", "-v", fmt.Sprintf("%s:/videos", dir), "linuxserver/ffmpeg:latest", "-v", "info", "-i", fmt.Sprintf("/videos/%s", file), "-f", "null", "-")
    output, err := cmd.CombinedOutput()
    if err != nil {
        fmt.Fprintf(f, "Corruption detected in: %s\n", file)
        fmt.Fprintf(f, "Error message: %s\n", string(output))
    } else {
        fmt.Fprintf(f, "No corruption detected in: %s\n", file)
    }
}
